src/docs/product/integrations/integration-platform/public-integration.mdx

```diff
@@ -0,0 +1,210 @@
+
+![A list of integrations that a user can install.](integration-directory.png)
+
+## OAuth Process
+
+After installation, if your user has approved all permissions, Sentry will generate a grant code and an installation ID. This information is sent by way of the `installation.created` webhook to the webhook URL specified in your configuration.
+
+However, if your integration has a redirect URL configured, the integration redirects the user’s browser to the configured URL with the grant code and installation ID in the query params.
+
+Start building your integration by implementing the redirect URL endpoint, `/setup` — typically where you exchange the grant code for a token that allows you to make API calls to Sentry:
+
+```python
+import requests
+from flask import redirect, request
+
+@app.route('/setup', methods=['GET'])
+def setup():
+    code = request.args.get('code')
+    install_id = request.args.get('installationId')
+
+    url = u'https://sentry.io/api/0/sentry-app-installations/{}/authorizations/'
+    url = url.format(install_id)
+
+    payload = {
+        'grant_type': 'authorization_code',
+        'code': code,
+        'client_id': 'your-client-id',
+        'client_secret': 'your-client-secret',
+    }
+
+    resp = requests.post(url, json=payload)
+    data = resp.json()
+
+    token = data['token']
+    refresh_token = data['refreshToken']
+    # ... Securely store the install_id, token and refresh_token in DB ...
+
+    return redirect('https://sentry.io/settings/')
+```
+
+## Token Exchange
+
+In the initial installation, you'll need the grant code given to you in either the installation webhook request or the redirect URL, in addition to your integration's client ID and client Secret:
+
+```python
+url = u'https://sentry.io/api/0/sentry-app-installations/{}/authorizations/'
+url = url.format(install_id)
+
+payload = {
+    'grant_type': 'authorization_code',
+    'code': code,
+    'client_id': 'your-client-id',
+    'client_secret': 'your-client-secret',
+}
+```
+
+Tokens expire after eight hours, so you'll need to refresh your tokens accordingly:
+
+```python
+url = u'https://sentry.io/api/0/sentry-app-installations/{}/authorizations/'
+url = url.format(install_id)
+
+refresh_token = retrieve_refresh_token_from_db(install_id)
+
+payload = {
+    'grant_type': 'refresh_token',
+    'refresh_token': refresh_token,
+    'client_id': 'your-client-id',
+    'client_secret': 'your-client-secret',
+}
+```
+
+The data you can expect back for both the initial grant code exchange and subsequent token refreshes is as follows:
+
+```json
+{
+  "id": "38",
+  "token": "ec48bf98637d44c294ead7566513686237e74ab67a074c64b3aaca2d93dbb8f1",
+  "refreshToken": "c866f154a65841638d44ee26364409b0a1a67bd642bd46e7a476f34f810712d6",
+  "dateCreated": "2019-08-07T20:25:09.870Z",
+  "expiresAt": "2019-08-08T04:25:09.870Z",
+  "state": null,
+  "application": null
+}
+```
+
+## Auth Tokens
```

Comment:
As per my other comment on the Integration Platform page, I think we can reduce the duplicate stuff by changing this to something like:

> ## Authentication Tokens
> Public integrations generate [Authentication Tokens](/integration-platform#authentication-tokens) after a successful [OAuth Installation](#installation)
> These tokens automatically expire every eight hours, meaning they must be refreshed manually.